### PR TITLE
sfwbar 1.0_beta13 -> 1.0_beta14

### LINKED ISSUE #319278: Update request: sfwbar 1.0_beta13 → 1.0_beta14
- Package name: sfwbar
- Latest released version: 1.0_beta14

- Current version on the unstable channel: 1.0_beta13
- Current version on the stable/release channel: 1.0_beta13

- [ ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
NotAShelf

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #319278

+ [Release notes](https://github.com/LBCrion/sfwbar/releases/tag/v1.0_beta14)

Updated location of package definition to `pkgs/by-name/`

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NotAShelf — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sf/sfwbar/package.nix [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+  json_c,
+  pkg-config,
+  gtk-layer-shell,
+  libpulseaudio,
+  libmpdclient,
+  libxkbcommon,
+  alsa-lib,
+  makeWrapper,
 }:
 stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (luftmensch-luftmensch) [reply pkgs/by-name/sf/sfwbar/package.nix]: Done, but I'm not too sure about the let in binding only for the `version`

[the remark this answers is not in the document]
Comment (NotAShelf) [reply pkgs/by-name/sf/sfwbar/package.nix]: still preferrable to rec

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
ofborg what are you smoking

**Review by NotAShelf — APPROVED** [review]

**Review by wegank — CHANGES_REQUESTED** [review]
Hash mismatches (both on ofborg and on my machine).

**Comment by luftmensch-luftmensch** [comment]
> Hash mismatches (both on ofborg and on my machine).

On which architecture?

**Comment by wegank** [comment]
x86_64-linux.

File: pkgs/by-name/sf/sfwbar/package.nix [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
+  pname = "sfwbar";
+  inherit version;
 
   src = fetchFromGitHub {
     owner = "LBCrion";
-    repo = pname;
+    repo = "sfwbar";
     rev = "v${version}";
-    hash = "sha256-7oiuTEqdXDReKdakJX6+HRaSi1XovM+MkHFkaFZtq64=";
+    hash = "sha256-4brP1SXaWq/L0D87rvlrWhLU1oFPSwNNxBSzRr4jsTM=";
```
Comment (wegank) [inline pkgs/by-name/sf/sfwbar/package.nix:28]: Could you try replacing this with `hash = ""` and run nix build again?

Comment (luftmensch-luftmensch) [reply pkgs/by-name/sf/sfwbar/package.nix:28]: Apparently it gives me `sha256-7oiuTEqdXDReKdakJX6+HRaSi1XovM+MkHFkaFZtq64=` as the correct hash. Dunno why before I got a different hash

Comment (wegank) [reply pkgs/by-name/sf/sfwbar/package.nix]: I assume that's because you have `version = "1.0_beta13";` instead of `version = "1.0_beta14";`...

Comment (luftmensch-luftmensch) [reply pkgs/by-name/sf/sfwbar/package.nix]: Didn't see it...

Maybe I forgot to add it while doing the editing

**Review by wegank — APPROVED** [review]

**Pull request merged**